db/src/test/kotlin/no/nav/helsearbeidsgiver/inntektsmelding/db/river/LagreImSkjemaRiverTest.kt

```diff
@@ -0,0 +1,215 @@
+package no.nav.helsearbeidsgiver.inntektsmelding.db.river
+
+import io.kotest.core.spec.style.FunSpec
+import io.kotest.datatest.withData
+import io.kotest.matchers.ints.shouldBeExactly
+import io.kotest.matchers.maps.shouldContainExactly
+import io.mockk.Runs
+import io.mockk.clearAllMocks
+import io.mockk.every
+import io.mockk.just
+import io.mockk.mockk
+import io.mockk.verify
+import io.mockk.verifySequence
+import kotlinx.serialization.builtins.serializer
+import kotlinx.serialization.json.JsonElement
+import kotlinx.serialization.json.JsonNull
+import no.nav.helse.rapids_rivers.testsupport.TestRapid
+import no.nav.helsearbeidsgiver.domene.inntektsmelding.deprecated.AarsakInnsending
+import no.nav.helsearbeidsgiver.domene.inntektsmelding.deprecated.Innsending
+import no.nav.helsearbeidsgiver.domene.inntektsmelding.v1.til
+import no.nav.helsearbeidsgiver.felles.BehovType
+import no.nav.helsearbeidsgiver.felles.EventName
+import no.nav.helsearbeidsgiver.felles.Key
+import no.nav.helsearbeidsgiver.felles.json.toJson
+import no.nav.helsearbeidsgiver.felles.json.toMap
+import no.nav.helsearbeidsgiver.felles.rapidsrivers.model.Fail
+import no.nav.helsearbeidsgiver.felles.test.mock.mockInnsending
+import no.nav.helsearbeidsgiver.felles.test.rapidsrivers.firstMessage
+import no.nav.helsearbeidsgiver.felles.test.rapidsrivers.sendJson
+import no.nav.helsearbeidsgiver.inntektsmelding.db.InntektsmeldingRepository
+import no.nav.helsearbeidsgiver.utils.json.toJson
+import no.nav.helsearbeidsgiver.utils.test.date.juli
+import java.util.UUID
+
+class LagreImSkjemaRiverTest :
+    FunSpec({
+        val testRapid = TestRapid()
+        val mockInntektsmeldingRepo = mockk<InntektsmeldingRepository>()
+
+        LagreImSkjemaRiver(mockInntektsmeldingRepo).connect(testRapid)
+
+        beforeTest {
+            testRapid.reset()
+            clearAllMocks()
+        }
+
+        context("inntektsmeldingskjema lagres") {
+            withData(
+                mapOf(
+                    "hvis ingen andre inntektsmeldingskjemaer er mottatt" to null,
+                    "hvis ikke duplikat av tidligere inntektsmeldingskjemaer" to
+                        mockInnsending().copy(
+                            egenmeldingsperioder = listOf(13.juli til 31.juli),
+                        ),
+                ),
+            ) { eksisterendeIm ->
+                every { mockInntektsmeldingRepo.hentNyesteInntektsmelding(any()) } returns null
```

Comment:
Her burde du også ha tester for når denne _ikke_ er `null`. Kanskje én når den er duplikat og én når den ikke er det?